server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java

```diff
@@ -112,6 +117,53 @@ private CurrentWsResponse toWsResponse(DbSession dbSession, String userLogin) {
     return OrganizationPermission.all()
       .filter(permission -> userSession.hasPermission(permission, defaultOrganizationUuid))
       .map(OrganizationPermission::getKey)
-      .collect(Collectors.toList());
+      .collect(toList());
   }
+
+  private CurrentWsResponse.Homepage findHomepageFor(DbSession dbSession, UserDto user) {
+    CurrentWsResponse.Homepage homepage = null;
+    if (isNotBlank(user.getHomepageType())) {
+      homepage = homepageOf(dbSession, user);
+    }
+    return homepage != null ? homepage : defaultHomepageOf();
+  }
+
+  private CurrentWsResponse.Homepage homepageOf(DbSession dbSession, UserDto user) {
+    String publicKey = getPublicKey(dbSession, user.getHomepageType(), user.getHomepageValue());
+    if (publicKey != null) {
+      return CurrentWsResponse.Homepage.newBuilder()
+        .setType(CurrentWsResponse.HomepageType.valueOf(user.getHomepageType()))
+        .setValue(publicKey)
+        .build();
+    }
+    return null;
+  }
+
+  private String getPublicKey(DbSession dbSession, String homepageType, String homepageKey) {
+    if (CurrentWsResponse.HomepageType.PROJECT.toString().equals(homepageType)) {
+      com.google.common.base.Optional<ComponentDto> dto = dbClient.componentDao().selectByUuid(dbSession, homepageKey);
+      if (dto.isPresent()) {
+        return dto.get().getKey();
+      } else {
+        return null;
+      }
+    }
+    if (CurrentWsResponse.HomepageType.ORGANIZATION.toString().equals(homepageType)) {
+      Optional<OrganizationDto> dto = dbClient.organizationDao().selectByUuid(dbSession, homepageKey);
+      if (dto.isPresent()) {
+        return dto.get().getKey();
+      } else {
+        return null;
+      }
+    }
+    return EMPTY;
```

Comment:
why return empty and in other cases null?

Reply:
EMPTY means no value is required as such it's the default value, null means i was expecting something but can't provide

Reply:
-> thwow illegal state